### PR TITLE
Feature/workbench session

### PR DESCRIPTION
**Pull request by zachhannum**
Adds a new target for `rstudio/workbench-session`, which is a docker image that will eventually replace `r-session-complete` for use with the new session init container feature.

It is essentially a copy/paste of r-session-complete with two major differences:

- Don't install Workbench/Session components
- Don't install Quarto (included in the session init container) and TinyTex

### CODE REVIEW

File: workbench-session/Dockerfile.ubuntu2204 [hunk 1]
```diff
@@ -26,10 +25,7 @@ RUN apt-get update \
     && rm -rf /var/lib/apt/lists/* 
 
 # Add Jupyter, Python, and Quarto to the PATH
-ENV PATH="/opt/python/jupyter/bin:/opt/python/bin:/opt/quarto/${QUARTO_VERSION}/bin:${PATH}"
-
-### Install TinyTeX using Quarto ###
-RUN $SCRIPTS_DIR/install_quarto.sh --install-tinytex --add-path-tinytex
+ENV PATH="/opt/python/jupyter/bin:/opt/python/bin:/usr/lib/rstudio-server/bin/quarto/bin:${PATH}"
```
[the remark this answers is not in the document]
Comment (zachhannum) [reply workbench-session/Dockerfile.ubuntu2204:28]: It does not. The discussion happened in `#workbench-dev` so I can't link it here, but I'll re-post my notes here:

 I don't think the TinyTex installation works at all on the existing r-session-complete images:

```
zach.hannum@session-0da293c9ab5a7f73c50c8-zachhannum---vs-code-sessionqxv4f:~$ quarto tools
[✓] Inspecting tools

Tool         Status            Installed     Latest  
chromium     Not installed     ---           869685  
tinytex      Not installed     ---           v2024.11
```

From root:
```
root@session-0da293c9ab5a7f73c50c8-zachhannum---vs-code-sessionqxv4f:~# which tlmgr
/usr/local/bin/tlmgr

root@session-0da293c9ab5a7f73c50c8-zachhannum---vs-code-sessionqxv4f:~# ls -al /usr/local/bin/tlmgr
lrwxrwxrwx 1 root root 37 Nov 21 12:18 /usr/local/bin/tlmgr -> /root/.TinyTeX/bin/x86_64-linux/tlmgr
zach.hannum@session-0da293c9ab5a7f73c50c8-zachhannum---vs-code-sessionqxv4f:~$ echo $PATH
/usr/lib/rstudio-server/bin/pwb-code-server/bin/remote-cli:/opt/python/jupyter/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

Back in the session:

```
zach.hannum@session-0da293c9ab5a7f73c50c8-zachhannum---vs-code-sessionqxv4f:~$ stat /root/.TinyTeX/bin/x86_64-linux/tlmgr
stat: cannot statx '/root/.TinyTeX/bin/x86_64-linux/tlmgr': Permission denied
```

Okay, so it's on the PATH but we don't actually have the correct permissions to use it.
tinytex doesn't really seem to want to behave as a system dependency, it's installer sticks it in the user's home directory by default.

IMO we should just leave tinytex off the base installation and let users install it to their home directory via quarto if they need it.

File: workbench-session/README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 # Supported tags and respective Dockerfile links
 
-* [`jammy`, `ubuntu2204`, `jammy-2024.09.1`, `ubuntu2204-2024.09.1`](https://github.com/rstudio/rstudio-docker-products/blob/main/r-session-complete/Dockerfile.ubuntu2204)
+* [`ubuntu2204-r4.4.1_4.3.3-py3.12.6_3.11.10`, `ubuntu2204-r4.4.1_4.3.3-py3.11.10_3.10.15`, `ubuntu2204-r4.4.0_4.3.3-py3.12.1_3.11.7`](https://github.com/rstudio/rstudio-docker-products/blob/main/workbench-session/Dockerfile.ubuntu2204)
```
[the remark this answers is not in the document]
Comment (zachhannum) [reply workbench-session/README.md:11]: I thought we had decided to follow the same tagging pattern as the [connect content images](https://hub.docker.com/r/rstudio/content-pro/tags), which don't use vanity tags?

[the remark this answers is not in the document]
Comment (bschwedler) [reply workbench-session/README.md:11]: Thanks! I didn't have that context.

File: workbench-session/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+## Overview
+
+Built images are available from the
+[rstudio/workbench-session](https://hub.docker.com/r/rstudio/workbench-session)
+repository on Docker Hub.
+
+These images include the following layers:
+
+* Base OS
+* System packages required for R, R packages, RStudio Professional Drivers, and Workbench Session Components
```
Comment (skyeturriff) [inline workbench-session/README.md:63]:
```suggestion
* System packages required for R, R packages, and RStudio Professional Drivers
```

File: workbench-session/test/goss.yaml
Comment (skyeturriff) [inline workbench-session/test/goss.yaml:1]: I think there was a note about ensuring the default python in the session images had pip, would that be worth verifying in these tests?

**Review by skyeturriff — APPROVED** [review]
lgtm, just a few comments!

File: docker-bake.hcl [hunk 4]
```diff
@@ -185,6 +185,10 @@ variable WORKBENCH_BUILD_MATRIX {
     }
 }
 
+variable WORKBENCH_SESSION_MATRIX {
+    default = PRO_BUILD_MATRIX
```
Comment (ianpittwood) [inline docker-bake.hcl:189]: Just curious, what's the reasoning behind inheriting all the base image builds?

File: workbench-session/.snyk
Comment (ianpittwood) [inline workbench-session/.snyk:1]: We may not need to exclude these seeing as they're expired ignores.

File: docker-bake.hcl (hunk 4 above)
Comment (zachhannum) [reply docker-bake.hcl:189]: Similar to [content-pro](https://hub.docker.com/r/rstudio/content-pro/tags), I didn't see a reason to not offer a few different combinations of R and Python versions for images. I'm happy to be overridden on that decision if it doesn't make sense, though.

Comment (ianpittwood) [reply docker-bake.hcl:189]: OK, I can understand that. I think this is fine for now. It's something we can take into account for future planning.

**Review by bschwedler — APPROVED** [review]
These changes overall look good.

I would like us to revisit the build matrix for these images at a later time, but having parity with the current `r-session-complete` images seems like a sensible path forward.

**Pull request merged**